main.go

```diff
@@ -35,10 +34,10 @@ func checkForErrorsAndExit(err error) {
 			logger.Println(err)
 		}
 		// exit with the underlying error code
-		var retCode int = 1
-		if exiterr, ok := errors.Unwrap(err).(*exec.ExitError); ok {
-			status := exiterr.Sys().(syscall.WaitStatus)
-			retCode = status.ExitStatus()
+		var retCode int
+		retCode, err := shell.GetExitCode(err)
+		if err != nil {
+			retCode = 1
```

Comment:
You should log the error here explicitly, otherwise, it won't be clear why the exit code is 1 instead of some underlying exit code returned by Terraform.

```go
logger.Printf("Error retrieving underlying exit code, so Terragrunt will exit with error code 1: %v", err)).
```